pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -570,8 +570,9 @@ def horovod_train(self, model):
 
         if torch.cuda.is_available() and self.on_gpu:
             # Horovod: pin GPU to local rank
-            torch.cuda.set_device(hvd.local_rank())
-            model.cuda(hvd.local_rank())
+            self.root_gpu = hvd.local_rank()
```

Comment:
shall it be rather protected `self._root_gpu` also add it to Trainer.__init__

Reply:
Moved the logic to the init method.  Seems there isn't a protected `_root_gpu`.  I was using the existing public `root_gpu` property set elsewhere, following this pattern: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/distrib_data_parallel.py#L348

Let me know if the new approach looks good.